### DISCUSSION TITLE
Potentially allow for hiding unlocked options?

### DISCUSSION THREAD

**Discussion by Kitfoxpup** [BrentFarris/TextGameMaker#40] [Ideas]
Say the writer wants to hide an option until something is discovered. Currently options can be disabled, but they can't be hidden. Like if the player needs to examine the area before they can see that there's a note on a desk, you don't want the Read Note On Desk option to be visible before the player is fully aware of the note being on the desk.

**Comment by BrentFarris** [comment]
Added in #46, the node now has a drop down select instead of a checkbox.
- `on` = available at start
- `lock` = locked until unlocked, also visible, just disabled
- `hide` = locked until unlocked, also hidden, so player doesn't see it until unlocked

A special note about `hide`, after it has been unlocked, it can't go back to being hidden. If you were to re-lock the option, it takes the `lock` state and is disabled but visible.

Tested with the following flow:
![image](https://user-images.githubusercontent.com/1002223/224591396-15b142f4-37ff-40d1-aba7-b96ff8de48ff.png)

**Reply by Kitfoxpup** [reply]
Makes sense to me, and looks great! :D thank you!
